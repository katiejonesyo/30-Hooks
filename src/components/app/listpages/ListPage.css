.link {
    text-decoration: 'underline';
    border: 'solid 10px blue';
  }