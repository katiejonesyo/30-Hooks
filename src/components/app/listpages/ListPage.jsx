import React from 'react';
import { Link } from 'react-router-dom';
import ListItem from '../list/ListItem';
import PropTypes from 'prop-types';

function ListPage({ characters }) {
  const charactersList = characters.map((character) => {
    return (
      <li key={character._id}>
        <Link to={`detail/${character._id}`}>
          <ListItem
            key={character._id}
            name={character.name}
            photoUrl={character.photoUrl}
          />
        </Link>
      </li>
    );
  });

  return <ul data-testid="characters">{charactersList}</ul>;
}

ListPage.propTypes = {
  characters: PropTypes.array,
};

export default ListPage;

