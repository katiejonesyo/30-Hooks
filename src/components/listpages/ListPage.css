body {
  background-color: rgb(223, 200, 182);
  text-align: center;
  align-items: center;
  margin-bottom: 10%;
} 

h4 {
  padding-bottom: 5%;

}

.link {
  text-decoration: 'none';
}

ul {
  text-decoration: 'none';
  display: 'none';
}

img {
  box-shadow: rgb(56, 30, 13) 0px 0px 40px 50px;
  margin-bottom: 10%;
}



