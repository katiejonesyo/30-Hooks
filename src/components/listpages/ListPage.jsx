import React from 'react';
import { Link } from 'react-router-dom';
import ListItem from '../list/ListItem';
import PropTypes from 'prop-types';
import styles from './ListPage.css';

function ListPage({ characters }) {
  const charactersList = characters.map((character) => {
    return (
      <main className={styles.main}>
      <li key={character._id}>
        <Link to={`detail/${character._id}`}>
          <ListItem
            key={character._id}
            name={character.name}
            photoUrl={character.photoUrl}
          />
        </Link>
      </li>
      </main>
    );
  });

  return <ul data-testid="characters">{charactersList}</ul>;
}

ListPage.propTypes = {
  characters: PropTypes.array,
};

export default ListPage;



