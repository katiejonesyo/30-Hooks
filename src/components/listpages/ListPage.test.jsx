import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ListPage from '../listpages/ListPage';


const characters = [
  {
    _id: '1',
    name: 'someName',
    photoUrl: 'somePhoto',
  },
  {
    _id: '2',
    name: 'someName',
    photoUrl: 'somePhoto',
  },
  {
    _id: '3',
    name: 'someName',
    photoUrl: 'somePhoto',
  },
];

describe('ListPage component', () => {
  afterEach(() => cleanup());
  it('renders list of characters', async() => {
    const { asFragment } = render(
      <MemoryRouter>
        <ListPage characters={characters} />
      </MemoryRouter>
    );
    waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    })
  });
});



