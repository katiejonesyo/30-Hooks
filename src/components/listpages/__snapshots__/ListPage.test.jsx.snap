// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage component renders list of characters 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <main
      class="main"
    >
      <li>
        <a
          href="/detail/1"
        >
          <main>
            <h1>
              someName
            </h1>
            <img
              alt="avatar character"
              class="image"
              src="somePhoto"
            />
          </main>
        </a>
      </li>
    </main>
    <main
      class="main"
    >
      <li>
        <a
          href="/detail/2"
        >
          <main>
            <h1>
              someName
            </h1>
            <img
              alt="avatar character"
              class="image"
              src="somePhoto"
            />
          </main>
        </a>
      </li>
    </main>
    <main
      class="main"
    >
      <li>
        <a
          href="/detail/3"
        >
          <main>
            <h1>
              someName
            </h1>
            <img
              alt="avatar character"
              class="image"
              src="somePhoto"
            />
          </main>
        </a>
      </li>
    </main>
  </ul>
</DocumentFragment>
`;
